### PR TITLE
Block during TLS handshake

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/1202279501986195/1203277901387351/f

### Description
Move the blocking point from SYN to TLS handshake

### Steps to test this PR
_Fresh install_
- [x] fresh install from this branch
- [x] enable AppTP
- [x] launch an app that tracks eg. speed test or reddit
- [x] trackers should inmediately be blocked

_Test app update_
- [x] install develop from develop and enable AppTP
- [x] update from this branch
- [x] ensure AppTP continues blocking as usual

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2482** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2482" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2482?utm_source=stack-comment).

File: app/src/test/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStackTest.kt [hunk 1]
```diff
@@ -148,64 +140,82 @@ class NgVpnNetworkStackTest {
...
+    fun whenIsDomainBlockedAndDomainIsFirstPartyTrackerThenReturnFalse() {
         val uid = 1200
+        val hostname = TEST_APP_TRACKER.hostname
 
-        whenever(appTrackerRepository.findTracker(eq(THIRD_PARTY_TRACKER.tracker.hostname), anyString())).thenReturn(THIRD_PARTY_TRACKER)
-        whenever(vpnAppTrackerBlockingDao.getRuleByTrackerDomain(THIRD_PARTY_TRACKER.tracker.hostname)).thenReturn(null)
-        ngVpnNetworkStack.onSniResolved(SniRR(THIRD_PARTY_TRACKER.tracker.hostname, "1.1.1.1"))
+        whenever(appTrackerRepository.findTracker(eq(hostname), anyString())).thenReturn(FIRST_PARTY_TRACKER)
+        whenever(vpnAppTrackerBlockingDao.getRuleByTrackerDomain(hostname)).thenReturn(null)
+        ngVpnNetworkStack.isDomainBlocked(DomainRR(hostname, uid))
```
Comment (nshuba) [inline app/src/test/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStackTest.kt:182]: Do we need this extra call to `isDomainBlocked`? I think we should be able to immediately assert that it returns true/false for this test and the ones below.

Comment (aitorvs) [reply app/src/test/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStackTest.kt:182]: oh, dumb error when I fixing the test. Will remove

**Review by nshuba — APPROVED** [review]
LGTM, thanks for fixing this!

**Pull request merged**